pkg/scalers/gcp_pub_sub_scaler.go

```diff
@@ -33,10 +39,12 @@ type pubsubScaler struct {
 }
 
 type pubsubMetadata struct {
-	targetSubscriptionSize int
-	subscriptionName       string
-	gcpAuthorization       gcpAuthorizationMetadata
-	scalerIndex            int
+	mode  string
+	value int
+
+	subscriptionName string
+	gcpAuthorization gcpAuthorizationMetadata
+	scalerIndex      int
```

Comment:
We should at least be backward compatible and deprecate the old field.

Reply:
You are right. I completely missed this one. We should:
1. if user specifies only `subscriptionSize`(the previous behavior) -> Print a warning message that this property is deprecated and users should use `mode` and `value` instead. But the scaler should work (we should internally set in this case `mode=SubscriptionSize` and populate `value` properly.
2. if user specifies `subscriptionSize` and at least one of `mode` or `value` -> return error that both cannot be set and user should use `mode` and `value` only.


@Friedrich42 could you please fix this? Thanks!